File: BHOI_/2013/Povezivanje gradova/grader.pas
Program grader;
uses gradovi;
const
   inbuf_len  = 1 shl 23;
   outbuf_len = 1 shl 10;
var
  n, i, j, x, y : LongInt;
  m : Array of LongInt;
  inbuf      : array [1..inbuf_len] of Byte;
  outbuf     : array [1..outbuf_len] of Byte;
begin
  { Set input and output buffering }
  settextbuf(input, inbuf);
  settextbuf(output, outbuf);

  readln(n);
  SetLength(m, n);
  for i := 0 to n-2 do
    read(m[i]);
  readln(m[n-1]);
  Inicijalizacija(n, m);
  for i := 0 to n-1 do
    for j := 0 to m[i]-1 do
    begin
      readln(x, y);
      PostaviKucu(i, x, y);
    end;
  writeln(Povezi());
end.

File: BHOI_/2013/Povezivanje gradova/grader.cpp
#include <cstdio>
#include <cassert>

using namespace std;

#define inbuf_len 1 << 23
#define outbuf_len 1 << 10

void Inicijalizacija(int n, int m[]);
void PostaviKucu(int i, int x, int y);
int Povezi();

int main() {
  int n, dummy;
  
  /* Set input and output buffering */
  char *inbuf, *outbuf;
  inbuf = new char[inbuf_len];
  outbuf = new char[outbuf_len];
  dummy = setvbuf(stdin, inbuf, _IOFBF, inbuf_len);
  assert(dummy == 0);
  dummy = setvbuf(stdout, outbuf, _IOFBF, outbuf_len);
  assert(dummy == 0);
  
  dummy = scanf("%d", &n);
  int *m = new int[n];
  for (int i = 0; i < n; i++)
    dummy = scanf("%d", &m[i]);
  Inicijalizacija(n, m);
  for (int i = 0; i < n; i++)
    for (int j = 0; j < m[i]; j++) {
      int x, y;
      dummy = scanf("%d %d", &x, &y);
      PostaviKucu(i, x, y);
    }
  dummy = printf("%d\n", Povezi());
  return 0;
}

